driver-mapping/src/main/java/com/datastax/driver/mapping/config/MappingConfiguration.java

```diff
@@ -0,0 +1,136 @@
+/*
+ *      Copyright (C) 2012-2015 DataStax Inc.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+package com.datastax.driver.mapping.config;
+
+/**
+ * The configuration to use for the mappers.
+ * <p/>
+ * The following strategies can be configured:
+ * <ul>
+ * <li>{@link PropertyAccessStrategy property access strategy}: defines how to access mapped properties.</li>
+ * <li>{@link PropertyTransienceStrategy property transience strategy}: defines whether or not a given property is transient.</li>
+ * <li>{@link HierarchyScanStrategy hierarchy scanning strategy}: defines how to scan for mapped properties in parent classes.</li>
+ * </ul>
+ */
+public class MappingConfiguration {
+
+    /**
+     * Returns a new {@link Builder} instance.
+     *
+     * @return a new {@link Builder} instance.
+     */
+    public static MappingConfiguration.Builder builder() {
+        return new MappingConfiguration.Builder();
+    }
+
+    /**
+     * Builder for {@link MappingConfiguration} instances.
+     */
+    public static class Builder {
+
+        private PropertyAccessStrategy propertyAccessStrategy = new DefaultPropertyAccessStrategy();
```

Comment:
This creates new default instances every time you construct a `MappingConfiguration`.  This may be an overoptimization, but you could have a static singleton for each strategies default implementation so we don't need to construct defaults every time.  On the other hand, creating `MappingConfiguration` should not be very frequent, so it might not be worth it.